Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -1936,12 +1936,41 @@ static bool insideImageOverlay(const VisiblePosition& position)
         clearSelection();
 }
 
+void WebPage::setShouldDrawVisuallyContiguousBidiSelection(bool value)
+{
+    if (m_shouldDrawVisuallyContiguousBidiSelection == value)
+        return;
+
+    m_shouldDrawVisuallyContiguousBidiSelection = value;
+
+    if (!value)
+        scheduleFullEditorStateUpdate();
```

Comment:
Why is an editor state update not needed when going false->true?

Reply:
Good catch — it's (technically) not needed at the moment because whenever we enter this mode, we're about to start changing the selection anyways which will schedule editor state updates.

However, I agree it makes sense to `scheduleFullEditorStateUpdate();` here, since we're not *always* guaranteed to trigger subsequent ES updates in the future. I'll remove the `!value` check here.